components/translate/core/browser/resources/brave_translate.js

```diff
@@ -0,0 +1,77 @@
+const securityOriginHost = new URL(securityOrigin).host;
```

Comment:
I think we should add a lot of comments into this script to let everyone understand how this whole machinery works.

Reply:
done.